lisa/tests/scheduler/eas_behaviour.py

```diff
@@ -341,17 +341,24 @@ def test_slack(self, negative_slack_allowed_pct=15) -> ResultBundle:
             activations with negative slack.
         :type negative_slack_allowed_pct: int
 
-        Use :class:`lisa.analysis.rta.PerfAnalysis` to find instances where the
-        RT-App workload wasn't able to complete its activations (i.e. its
-        reported "slack" was negative). Assert that this happened less than
+        Use :class:`lisa.analysis.rta.EASEventsAnalysis` to find instances
+        where the RT-App workload wasn't able to complete its activations (i.e.
+        its reported "slack" was negative). Assert that this happened less than
         ``negative_slack_allowed_pct`` percent of the time.
         """
-        analysis = PerfAnalysis.from_dir(self.res_dir)
-
         passed = True
         bad_activations = {}
-        for task in analysis.tasks:
-            slack = analysis.get_df(task)["Slack"]
+        skip_main = True
```

Comment:
Can be simplified into:
```
tasks = self.trace.analysis.rta.rtapp_tasks
# remove() only removes the first occurence
tasks.remove(TaskID(None, 'rt-app'))
if any(task.comm == 'rt-app' for task in tasks):
     raise ValueError('Test misconfigured, rt-app tasks cannot be named "rt-app"')
```

Reply:
Actually, it seems that this check should be better done in `RTAEventsAnalysis.__init__` rather than in this specific test. That said, there are legitimate cases where the trace could contain more than one rt-app task.

A better fix would be ensuring that `rt-app` is not used as a task name when creating the profile, which will remove the need to check it here since we control the profile. In the general case, we cannot assume implement the check since there is ambiguity, so people should just not do that. (maybe rt-app itself should forbid that)

Reply:
Looks like `tasks.remove(TaskID(None, 'rt-app'))` fails since we cannot find a TaskID with a None PID. Moreover, if by any chance the main task is not there (e.g. we use a trace view) that remove will fails again.
Finally, IMHO the code above is not "simpler"... it's just different. ;)

What we have in the patch now works and it performs a validity check while we run the tests.
So, if that's your only final comment, I would suggest to keep it as it is.

Reply:
My final comment is the second part of the comment:

> A better fix would be ensuring that rt-app is not used as a task name when creating the workload from a profile, which will remove the need to check it here since we control the profile. In the general case, we cannot assume implement the check since there is ambiguity, so people should just not do that. (maybe rt-app itself should forbid that)

There is no real reason why `test_slack()` should be blessed and the rest of the codebase not benefit from the check if it always makes sense. If it would reject valid traces in some cases, a much better fix is to prevent that situation from happening straight at the source in wlgen before rt-app even runs, which will benefit all rt-app workloads.

Reply:
In principle, there could be tests where you still want to access the main task, why now?!
Thus, I see this filtering as something test-specific. You just need to know that `trace.analysis.rta.rtapp_tasks` returns *all* the rt-app tasks and code your code accordingly.

Otherwise, we can factorize the filtering code into the RTA analysis and always and only return tasks which are not the main task.

But honestly, this has nothing to do with the current API being added by this PR and the code in here maintains the tests working as expected. Future changes are welcome as always and thanks Good we are not running out or PR numbers ;)